admin-dev/themes/new-theme/js/pages/product/components/categories/category-tree-selector.js

```diff
@@ -24,71 +24,82 @@
  */
 
 import Bloodhound from 'typeahead.js';
-import _ from 'lodash';
 
 import AutoCompleteSearch from '@components/auto-complete-search';
 import Tokenizers from '@components/bloodhound/tokenizers';
 import ProductMap from '@pages/product/product-map';
 import ProductEventMap from '@pages/product/product-event-map';
 import {getCategories} from '@pages/product/services/categories';
+import Tags from '@pages/product/components/categories/tags';
 
 const {$} = window;
 
 const ProductCategoryMap = ProductMap.categories;
 
-export default class CategoriesManager {
-  /**
-   * @param {EventEmitter} eventEmitter
-   * @returns {{}}
-   */
+export default class CategoryTreeSelector {
   constructor(eventEmitter) {
     this.eventEmitter = eventEmitter;
-    this.categoriesContainer = document.querySelector(
-      ProductCategoryMap.categoriesContainer,
-    );
     this.categories = [];
     this.typeaheadDatas = [];
-    this.categoryTree = this.categoriesContainer.querySelector(ProductCategoryMap.categoryTree);
-    this.prototypeTemplate = this.categoryTree.dataset.prototype;
-    this.prototypeName = this.categoryTree.dataset.prototypeName;
-    this.expandAllButton = this.categoriesContainer.querySelector(ProductCategoryMap.expandAllButton);
-    this.reduceAllButton = this.categoriesContainer.querySelector(ProductCategoryMap.reduceAllButton);
 
-    this.initCategories();
+    return {
+      showModal: (selectedCategories, defaultCategoryId) => this.showModal(selectedCategories, defaultCategoryId),
+    };
+  }
+
+  showModal(selectedCategories, defaultCategoryId) {
+    this.selectedCategories = selectedCategories;
+    this.defaultCategoryId = defaultCategoryId;
+    const modalContent = $(ProductCategoryMap.categoriesModalTemplate);
+    // @todo: replace fancybox with Modal after following PR is merged - https://github.com/PrestaShop/PrestaShop/pull/25184
+    $.fancybox({
+      type: 'iframe',
+      width: '90%',
+      height: '90%',
+      fitToView: false,
+      autoSize: false,
+      content: modalContent.html(),
+      afterShow: () => {
+        this.initCategories();
+      },
+    });
+  }
 
-    return {};
+  onApplyCategoryChanges() {
```

Comment:
```suggestion
  listenForCategoryChanges() {
```

At first I thought this was called to trigger the event, but the purpose here is to watch for click event